src/Symfony/Bridge/Doctrine/Form/ChoiceList/IdReader.php

```diff
@@ -0,0 +1,125 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bridge\Doctrine\Form\ChoiceList;
+
+use Doctrine\Common\Persistence\Mapping\ClassMetadata;
+use Doctrine\Common\Persistence\ObjectManager;
+use Symfony\Component\Form\Exception\RuntimeException;
+
+/**
+ * A utility for reading object IDs.
+ *
+ * @since  1.0
+ * @author Bernhard Schussek <bschussek@gmail.com>
+ *
+ * @internal This class is meant for internal use only.
+ */
+class IdReader
+{
+    /**
+     * @var ObjectManager
+     */
+    private $om;
+
+    /**
+     * @var ClassMetadata
+     */
+    private $classMetadata;
+
+    /**
+     * @var bool
+     */
+    private $singleId;
+
+    /**
+     * @var bool
+     */
+    private $intId;
+
+    /**
+     * @var string
+     */
+    private $idField;
+
+    public function __construct(ObjectManager $om, ClassMetadata $classMetadata)
+    {
+        $ids = $classMetadata->getIdentifierFieldNames();
+        $idType = $classMetadata->getTypeOfField(current($ids));
+
+        $this->om = $om;
+        $this->classMetadata = $classMetadata;
+        $this->singleId = 1 === count($ids);
+        $this->intId = $this->singleId && 1 === count($ids) && in_array($idType, array('integer', 'smallint', 'bigint'));
```

Comment:
``1 === count($ids)`` is useless here. It is already checked by ``$this->singleId``